java/memory/src/main/java/org/apache/arrow/memory/util/hash/NonNegativeHasher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.memory.util.hash;
+
+/**
+ * Wrapper for another {@link ArrowBufHasher}, to make the generated
```

Comment:
I don't think this is super useful.  For hash-tables, I expect the implementation would be masking the result incrementally anyways to achieve positive values.  Are there other use-cases you had in mind?

Reply:
Yes. A use case is here:
https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/util/MathUtils.java

In Flink, all hash codes are supposed to be positive.

Integer masking is a better choice for conditional negation.